src/components/layout/nav/search/Search.jsx

```diff
@@ -1,7 +1,156 @@
+      "https://fgl.scene7.com/is/image/FGLSportsLtd/FGL_332654173_01_a-Nike-Mens-Flex-Control-3-Training-Shoes-Black-Anthracite-AJ5911-002?wid=800&hei=800&bgColor=0,0,0,0&fmt=png-alpha&resMode=sharp2&op_sharpen=1",
+      "https://fgl.scene7.com/is/image/FGLSportsLtd/FGL_332762662_10_a-Nike-Womens-Air-Precision-II-Basketball-Shoes-White-Black-AA7069-100?bgColor=0,0,0,0&fmt=png-alpha&hei=528&resMode=sharp&qlt=85,1&op_sharpen=1",
+    ],
+  },
+  {
+    title: "V2 ZYON",
+    brand: "Adidas",
+    price: 250,
+    currency: "$",
+    tags: ["shoe", "shoes", "adidas", "v2", "zyon"],
+    images: [
+      "https://fgl.scene7.com/is/image/FGLSportsLtd/FGL_332979914_01_a?wid=800&hei=800&bgColor=0,0,0,0&fmt=png-alpha&resMode=sharp2&op_sharpen=1",
+      "https://fgl.scene7.com/is/image/FGLSportsLtd/FGL_332654173_01_a-Nike-Mens-Flex-Control-3-Training-Shoes-Black-Anthracite-AJ5911-002?wid=800&hei=800&bgColor=0,0,0,0&fmt=png-alpha&resMode=sharp2&op_sharpen=1",
+      "https://fgl.scene7.com/is/image/FGLSportsLtd/FGL_332762662_10_a-Nike-Womens-Air-Precision-II-Basketball-Shoes-White-Black-AA7069-100?bgColor=0,0,0,0&fmt=png-alpha&hei=528&resMode=sharp&qlt=85,1&op_sharpen=1",
+    ],
+  },
+];
 
 const Search = () => {
-  return <div></div>;
+  // It is left empty because we want search to start as empty.
+  const [searchQueryContainer, setSearchQuery] = useState("");
+  const [showEmptySearchWarning, setEmptySearchWarning] = useState(false);
+
+  const onlyFirstImageOfProduct = 0;
+  const amountOfProductToShowInSearch = 3;
+
+  // This will be sending input value to database with fetching if input is not empty when "Search" button is clicked.
+  const sendSearchQuery = () => {
+    if (!searchQueryContainer) {
+      setEmptySearchWarning(true);
+      return;
+    } else {
+      setEmptySearchWarning(false);
+    }
+  };
+
+  // This will setting input values onChange and will show preview card only if there is something typed.
+  const handleSearchChanges = (e) => {
+    setSearchQuery(e.target.value);
+    if (!searchQueryContainer) {
+      setEmptySearchWarning(true);
+      return;
+    } else {
+      setEmptySearchWarning(false);
+    }
+  };
+
+  const handleSearchPreview = () => {
+    const previewItemsContainer = mockDatabase
+      .filter((product) => product.tags.includes(searchQueryContainer))
+      .slice(0, amountOfProductToShowInSearch)
+      .map((product, idx) => (
+        <Col xl={3} lg={3} md={3} className="card" key={idx}>
+          <img src={product.images[onlyFirstImageOfProduct]} alt="product" />
+          <div className="title">{product.title}</div>
+          <div className="brand">{product.brand}</div>
+          <div className="price">
+            {product.currency}
+            {product.price}
+          </div>
+        </Col>
+      ));
+
+    // This will be more precise to show if it's still loading or failed as searching when we implement Fetching.
```

Comment:
use TODO(github name) here too for any incomplete code, in general and in the future